src/server/req_manager.c

```diff
@@ -1846,6 +1845,36 @@ mgr_sched_unset(struct batch_request *preq)
 				svr_save_db(&server, SVR_SAVE_FULL);
 				free_svrattrl(t_list);
 			}
+		}  else if (strcasecmp(tmp_plist->al_name, ATTR_sched_threads) == 0) {
+			svrattrl *t_list;
+			int num_cores;
+
+			/* By default, set sched_threads to the number of cores on the machine */
+
+#ifdef WIN32
+			SYSTEM_INFO sysinfo;
+			GetSystemInfo(&sysinfo);
+			num_cores = sysinfo.dwNumberOfProcessors;
+#else
+			num_cores = sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
It is probably going to make it too complicated but what happens where there are multiple-schedulers involved? would it set it to the same value for all scheds?
If you don't want to change it, then at least move this assignment to scheduler because scheduler might end up running on a machine where the server isn't present, so the cores won't match.